src/server/server.rs

```diff
@@ -475,7 +476,7 @@ mod tests {
             })
             .unwrap();
 
-        rx.recv().unwrap();
+        assert!(rx.recv_timeout(Duration::from_secs(5)).is_ok());
```

Comment:
The server thread will not panic if it doesn't receive messages. the server shutdown is skipped in test case if this assertion is triggered.